pulp_smart_proxy/app/views.py

```diff
@@ -69,5 +69,10 @@ class VersionView(APIView):
         operation_id="version_read",
     )
     def get(self, request):
-        data = {"version": get_plugin_config("smart_proxy").version}
+        data = {
```

Comment:
This is where I thought we could extend the versions
```suggestion
        data = {
            "server": "pulpcore",
```

And then add `foreman-proxy` in the Ruby implementation.

Then Foreman can know whether to apply version checks or not.

Reply:
https://github.com/theforeman/foreman/blob/f9f11624ae7898092addc9f2565657ab2db6ec9e/app/controllers/smart_proxies_controller.rb#L150-L170 is the code for that version check.

Reply:
I like that! Not sure I'd call that `server`, or `application` or `implementation`, but we can do the bike-shedding later!

Reply:
cc @shimshtein this may also be relevant for you with IoP. IIRC you even wrote the original version checks.

The problem is that Foreman checks for version mismatches. My idea was that we should only apply those checks if it's `foreman-proxy` because other implementations (like this one or IoP) will follow different versioning schemes.

Another option is to completely drop the version checks and fully adopt the pattern where we check for capabilities (https://theforeman.org/2019/04/smart-proxy-capabilities-explained.html) instead of versions.